compiler/rustc_codegen_llvm/src/coverageinfo/mapgen.rs

```diff
@@ -26,7 +26,10 @@ use tracing::debug;
 /// undocumented details in Clang's implementation (that may or may not be important) were also
 /// replicated for Rust's Coverage Map.
 pub fn finalize<'ll, 'tcx>(cx: &CodegenCx<'ll, 'tcx>) {
-    let function_coverage_map = cx.coverage_context().take_function_coverage_map();
+    if cx.coverage_context().is_none() {
+        return;
+    }
+    let function_coverage_map = cx.coverage_context().unwrap().take_function_coverage_map();
```

Comment:
Nit: Something like this might be a little better style-wise. 

```suggestion
    let function_coverage_map = match cx.coverage_context() {
        Some(ctx) => ctx.take_function_coverage_map(),
        None => return,
    };
```

Reply:
Agreed. Thanks!